### PR TITLE
protos: Update style guide to try and prevent redundant imports

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:  protos: Update style guide to try and prevent redundant imports
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] Partial fix for #15589
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
@htuch similar to #16231 im really not too sure what exactly im doing here - but like that PR im  trying to get the ball rolling...

**Review by htuch — COMMENTED** [review]
LGTM. Why not verify this works as expected on some extension?

File: api/STYLE.md [hunk 1]
```diff
@@ -122,8 +122,7 @@ To add an extension config to the API, the steps below should be followed:
        deps = ["@com_github_cncf_udpa//udpa/annotations:pkg"],
    )
    ```
-1. Add to the v3 extension config proto `import "udpa/annotations/migrate.proto";`
-   and `import "udpa/annotations/status.proto";`
```
Comment (phlax) [inline api/STYLE.md:126]: im guessing in this case that its ok to just totally remove this

File: api/STYLE.md [hunk 2]
```diff
@@ -122,8 +122,7 @@ To add an extension config to the API, the steps below should be followed:
...
+1. Add to the v3 extension config proto `import "udpa/annotations/status.proto";`
```
Comment (phlax) [inline api/STYLE.md:125]: not sure about this one - this is also added and unused frequently - but im guessing it is needed (in some cases ?) and may relate to some of the docs below ?

Comment (htuch) [reply api/STYLE.md:125]: This is needed for the `work_in_progress` annotation below.

Comment (phlax) [reply api/STYLE.md:125]: k - on that basis lets keep - i think the issue is more status not being set than the status proto being included

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**